### PR TITLE
ERC2477 implementation

### PR DESCRIPTION
**Pull request by MoMannn**

### CODE REVIEW

File: src/tests/nf-token-metadata-integrity.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const { expect } = require('chai');
+
+describe('nf-token-metadata-integrity', function() {
+  let nfToken, owner, bob;
+  const schemaDigest = '0x9c22ff5f21f0b81b113e63f7db6da94fedef11b2119b4088b89664fb9a3cb658';
```
Comment (xpepermint) [inline src/tests/nf-token-metadata-integrity.js]: I'd be happy to see `digest` values to be created from actual source included in the test case.

File: src/contracts/erc2477.sol [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.6;
+/**
+ * @title ERC-2477 Non-Fungible Token Metadata Integrity Standard
```
Comment (xpepermint) [inline src/contracts/erc2477.sol:5]: @fulldecent what do you think? Is it ok if we include `Non-Fungible` here? Should this be in parenthesis?

File: src/contracts/erc2477.sol [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.6;
```
Comment (xpepermint) [inline src/contracts/erc2477.sol:3]: Let's be on the edge `0.8.7`.

Comment (MoMannn) [reply src/contracts/erc2477.sol:3]: erc721 is on 0.8.6.

File: src/tests/nf-token-metadata-integrity.js (hunk 1 above)
Comment (MoMannn) [reply src/tests/nf-token-metadata-integrity.js]: Done. Check if ok.

**Pull request merged**